zeus/metric.py

```diff
@@ -0,0 +1,532 @@
+"""Track and export energy and power metrics via Prometheus."""
+
+from __future__ import annotations
+
+import abc
+import time
+import warnings
+import multiprocessing as mp
+
+from typing import Sequence
+
+from prometheus_client import (
+    CollectorRegistry,
+    Histogram,
+    Counter,
+    Gauge,
+    push_to_gateway,
+)
+
+from zeus.monitor.power import PowerMonitor
+from zeus.monitor.energy import ZeusMonitor
+
+from zeus.device.cpu import get_cpus
+
+
+class Metric(abc.ABC):
+    """Abstract base class for all metric types in Zeus.
+
+    Defines a common interface for metrics, ensuring consistent behavior
+    for `begin_window` and `end_window` operations.
+    """
+
+    @abc.abstractmethod
+    def begin_window(self, name: str):
```

Comment:
Both `begin_window` and `end_window` must accept `sync_execution: bool = True`, following `ZeusMonitor`. Then, `sync_execution` should be passed in as is to `ZeusMonitor.begin_window` or `ZeusMonitor.end_window`.